### PR TITLE
Have applications read in config from custom input

### PR DESCRIPTION
**Pull request by yfei1**
Per discussion, we will have an evaluator and mmf live under the `internal/testing` or `test/e2e` folder for testing purpose. Therefore, it make sense for these components to reuse the internal library instead of reinventing the wheels.

However, the internal config library requires its consumers read in two different config files from different location which complicated the evaluator and mmf setup.

This commit changed the `app.RunApplication` interface to read configurations from a custom getConfig function, such that the new evaluator and mmf could fake their configs and decouple themselves from the internal config library.

### CODE REVIEW

**Comment by Laremere** [comment]
What is the complexity of the config that needs to be passed?  Eg, are the log, metric, etc settings going to need to be duplicated?

**Comment by yfei1** [comment]
> What is the complexity of the config that needs to be passed? Eg, are the log, metric, etc settings going to need to be duplicated?

For the evaluator, we need the hostname, grpcport and httpport at mininum.
For the mmf, we need the three fields above and a mmlogic address.

**Comment by Laremere** [comment]
That's not all app.RunApplication uses though:

logging.ConfigureLogging(cfg)
rpc.NewServerParamsFromConfig(cfg,

NewServerParamsFromConfig uses ConfigNameEnableRPCLogging, telemetry.ConfigNameEnableMetrics, etc.

**Comment by yfei1** [comment]
> That's not all app.RunApplication uses though:
> 
> logging.ConfigureLogging(cfg)
> rpc.NewServerParamsFromConfig(cfg,
> 
> NewServerParamsFromConfig uses ConfigNameEnableRPCLogging, telemetry.ConfigNameEnableMetrics, etc.

These configs have their default values set to false. IMO, perhaps we don't necessary need the rpc logging and payload logging for an end-to-end test.

File: internal/app/appmain.go [hunk 1]
```diff
@@ -30,8 +30,8 @@ var (
 )
 
 // RunApplication creates a server.
-func RunApplication(serverName string, bindService func(*rpc.ServerParams, config.View) error) {
-	cfg, err := config.Read()
```
Comment (sawagh) [inline internal/app/appmain.go]: I see the issue here  (or even in the existing implementation) of the RunApplication depending on values that may just not be set. 

One approach could be for it to define explicitly, a struct with the values it expects - and the callers to fill in that before calling RunApplication. Atleast from readability perspective we could then have a clear understanding of the values the caller chose to not to populate and document its impact.

Wont block this PR for it but can you create an issue for refactoring this?

Comment (yfei1) [reply internal/app/appmain.go]: Done.

**Pull request merged**